### ISSUE TITLE
Nvidia Drivers ahead of flatpak

### ISSUE THREAD

**Issue by dhoell** [ublue-os/hwe#275]
The current nvidia 560 drivers are still labeled "beta" ([https://www.nvidia.com/en-us/drivers/unix/](https://www.nvidia.com/en-us/drivers/unix/)), thus are not available on flathub. Negativo17 however still packages them, causing all flatpaks to lose nvidia driver functionality, as they are stuck on 555.

**Comment by ManuaL46** [comment]
Yes this issue popped up with 20240808 and 20240809 images, for some reason I have the 560 client driver version but the kernel module is still using the 555 driver version.

Has anyone else faced this issue, currently sticking with 20240807 fixes the issue.

**Comment by castrojo** [comment]
Please check again, we found the issue and corrected it. Builds just finished now.

**Comment by castrojo** [comment]
Oh sorry I misspoke, we were correcting the skew between this repo and akmod to fix having mixed 555/560 on the same image, stay pinned on your existing image for now.

**Comment by Pryka** [comment]
> The current nvidia 560 drivers are still labeled "beta" (https://www.nvidia.com/en-us/drivers/unix/), thus are not available on flathub. Negativo17 however still packages them, causing all flatpaks to lose nvidia driver functionality, as they are stuck on 555.

Try updating image and rerunning `flatpak upgrade` flathub repo already have 560 beta drivers on it. There is a chance that was not downloaded due to mixed kernel module and drivers version.

If this not help, try enabling flathub beta repository.

**Comment by dhoell** [comment]
As far as I understand this is about a mismatch between Nagativo17 and Flathub repositories. Negativo17 publishes nvidia 560, which is still beta according to upstream. Which is why other repositories (like flathub) don't publish it in their release branch, which is IMO absolutely correct, and Negativo17 should not publish beta software in Fedora 40 release (rawhide, sure, but not in a stable release).

I'm sorry, I don't know about the reasons for the switch to Negativo17 repo, but if it was only speed of updates, I'd say that slower updates from RPMfusion are vastly preferable to fast, but premature and breaking updates. 

Yes, this can be fixed by getting flatpak nvidia drivers from flathub-beta, but "we may use beta drivers at any random time, so switch so flathub-beta when that happens" does not seem like an appropriate solution.

**Comment by Pryka** [comment]
> Yes, this can be fixed by getting flatpak nvidia drivers from flathub-beta, but "we may use beta drivers at any random time, so switch so flathub-beta when that happens" does not seem like an appropriate solution.

Enabling beta flatpak don't switch all apps to beta, you need to explicitly install them with `latpak install  flathub-beta app/runtime name`

It can be left enabled at all times.

**Comment by ManuaL46** [comment]
Hi this issue seems to have cropped up again, the system drivers and the flatpak drivers are different, and it seems that flathub stable still doesn't have this version

![image](https://github.com/user-attachments/assets/d144d976-7f25-41c1-b347-be32fc4726e0)

**Comment by Pryka** [comment]
> Hi this issue seems to have cropped up again, the system drivers and the flatpak drivers are different, and it seems that flathub stable still doesn't have this version
> 
> ![image](https://private-user-images.githubusercontent.com/119213535/360792781-d144d976-7f25-41c1-b347-be32fc4726e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MTMxMTcsIm5iZiI6MTcyNDQxMjgxNywicGF0aCI6Ii8xMTkyMTM1MzUvMzYwNzkyNzgxLWQxNDRkOTc2LTdmMjUtNDFjMS1iMzQ3LWJlMzJmYzQ3MjZlMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyM1QxMTMzMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMjg0NWFmN2U3ODExNzhkN2Y1MjUyZWVlOGViNzQwMDdlYTdmYTdiZDNlYTQxZTk0ZmJjZjgzYjgyY2E0MDYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZpUj9MO8hDQG4V3PMbap_IMrTkjzfXXgFiAS_fFKg-s)

Looks like people from flatpak package have some issues - https://github.com/flathub/org.freedesktop.Platform.GL.nvidia/commit/c8df268595409665615fdf411de03c8a1d190951

**Comment by seras42** [comment]
Any solutions to @ManuaL46 problem? Rerolling does not seem to help. This shutdown a lot of flatpak apps who run on nvidia. I can not even run VLC.

**Comment by ManuaL46** [comment]
> Any solutions to @ManuaL46 problem? Rerolling does not seem to help. This shutdown a lot of flatpak apps who run on nvidia. I can not even run VLC.

Rolling back to 2024.08.20 Fixed this for me the issue arises from 2024.08.21 onwards for me I think.

**Comment by bsherman** [comment]
Anecdotal evidence... negativo17 updated to 565 (also beta) just a few days ago, and I don't know when flatpak caught up, but:
```
anguirel in ~ took 3s 
$ rpm -q kmod-nvidia
kmod-nvidia-565.57.01-1.fc40.x86_64

anguirel in ~ took 4s 
$ flatpak list --all |grep nvidia
nvidia-565-57-01	org.freedesktop.Platform.GL.nvidia-565-57-01		1.4	system
```
